lib/utils/tunnel.ts

```diff
@@ -96,11 +97,14 @@ const openPortThroughProxy = (
 	}
 
 	return new Promise<Socket>((resolve, reject) => {
-		const proxyTunnel = new Socket();
-		proxyTunnel.on('error', reject);
-		proxyTunnel.connect(proxyPort, proxyServer, () => {
-			const proxyConnectionHandler = (data: Buffer) => {
-				proxyTunnel.removeListener('data', proxyConnectionHandler);
+		const proxyTunnel = tls.connect(proxyPort, proxyServer, {}, () => {
+			if (!proxyTunnel.authorized) {
```

Comment:
Is this test necessary? The documentation says that the `rejectUnauthorized` property defaults to `true`:
> https://nodejs.org/docs/latest-v10.x/api/tls.html#tls_tls_connect_options_callback
> `rejectUnauthorized` <boolean> If not `false`, the server certificate is verified against the list of supplied CAs. An `'error'` event is emitted if verification fails; `err.code` contains the OpenSSL error code. **Default:** `true`.

The documentation mentions a "list of supplied CAs", but it also applies to the ``tlsSocket.authorized`` property:

> https://nodejs.org/docs/latest-v10.x/api/tls.html#tls_tlssocket_authorized
> Returns true if the peer certificate was signed by one of the CAs specified when creating the tls.TLSSocket instance, otherwise false.

> https://nodejs.org/docs/latest-v10.x/api/tls.html#tls_event_secureconnection
> The tlsSocket.authorized property is a boolean indicating whether the client has been verified by one of the supplied Certificate Authorities for the server. If tlsSocket.authorized is false, then socket.authorizationError is set to describe how authorization failed. Note that depending on the settings of the TLS server, unauthorized connections may still be accepted.

I assume that in the absence of an explicit list of CAs, it uses Node's default / system default / environment variables.

Reply:
...from what I read, I think it is.

> The 'secureConnect' event is emitted after the handshaking process for a new connection has successfully completed. The listener callback will be called regardless of whether or not the server's certificate has been authorized. It is the client's responsibility to check the tlsSocket.authorized property to determine if the server certificate was signed by one of the specified CAs. If tlsSocket.authorized === false, then the error can be found by examining the tlsSocket.authorizationError property. If ALPN was used, the tlsSocket.alpnProtocol property can be checked to determine the negotiated protocol.